generator/src/lib.rs

```diff
@@ -1185,38 +1213,27 @@ pub fn generate_enum<'a>(
 }
 
 pub fn generate_result(ident: Ident, _enum: &vkxml::Enumeration) -> Tokens {
-    let notation = _enum.elements.iter().filter_map(|elem| {
-        let (variant_name, notation) = match *elem {
-            vkxml::EnumerationElement::Enum(ref constant) => (
-                constant.name.as_str(),
-                constant.notation.as_ref().map(|s| s.as_str()).unwrap_or(""),
-            ),
+    let display_notation = _enum.elements.iter().filter_map(|elem| {
+        let variant_name = match *elem {
+            vkxml::EnumerationElement::Enum(ref constant) => constant.name.as_str(),
             _ => {
                 return None;
             }
         };
 
         let variant_ident = variant_ident(&_enum.name, variant_name);
         Some(quote! {
-            #ident::#variant_ident => Some(#notation)
+            #ident::#variant_ident => Some(stringify!(#variant_ident))
```

Comment:
I stand by my judgement that this is a step backwards. Especially for error enums, `Debug` impls should display symbolic names (where available), and `Display` impls should display written descriptions for human-friendly error output. The descriptions here are taken directly from the Vulkan spec and should be plenty googlable, and will be more convenient in the common case of debugging.

Consistency is nice, but it's not sufficient reason to introduce a regression. If anything, we should move other parts of the system towards the more idiomatic pattern, though the idiom is most well-established for error enums and `Result` is the only one of those.

Reply:
The doc comments from the spec are in the doc comments of the generated output. I disagree that this is in any way a "regression". 

The error component of the result is an enum. When I print the enum I want to know the name of the enum, not a sentence describing it. The debug impls _do not print the name of the enum_, they print the struct type and the containing integer. This is helpful in some cases is, but there absolutely needs to be a way to see the **name** of the enum.

Reply:
> When I print the enum I want to know the name of the enum, not a sentence describing it

In practically all Rust code, this is what `Debug` impls on enums do; in particular, this is the behavior of a derived `Debug` impl on an enum.

> The debug impls do not print the name of the enum, they print the struct type and the containing integer

This is easily corrected; users who actually need the integer can still access it with `as_raw`.

Reply:
These are not *rust* enums, they're unit structs. The formatting behavior is different.

Reply:
They're newtype structs whose function is to emulate enums across an ABI that doesn't guarantee a closed set of possible values. In other words, they're enums except that they may have some statically unknown cases.

Reply:
Can you give me a diff of how the new output vs the old output would look like?

Reply:
https://github.com/MaikKlein/ash/pull/194/files#diff-06980beb7c96424568ed38b471ee57e9L40632

Reply:
I think it is already implemented correctly. Display is for userfacing output, I have to admit that I did not follow this rule this when I started ash. Still I believe we could improve the Debug output though. `Result(0)` is not super useful in practice. I think we should add something like `Result::Success(0)`, if possible. @Ralith Thoughts?

Reply:
I think the ideal approach is to match on the code and display the name if known, and the numeric value otherwise, similar to what's done in the `Display` impls of other Vulkan enums. I think I made a mistake in making those be `Display` impls in the first place, considering rust conventions.

Reply:
Do you want me to just pull out the changes with `Display` and `Result` so that it can be handled later?

Reply:
Yeah I think that would be best, we address this later.

Reply:
The `Display` changes were reverted in https://github.com/MaikKlein/ash/pull/194/commits/d52f2c9b3e59ddde9ee44f385210327d99392689

Is there anything else that needs to be changed?